turbopack/crates/turbopack-ecmascript/src/references/mod.rs

```diff
@@ -1207,6 +1207,28 @@ pub(crate) async fn analyse_ecmascript_module_internal(
                                 .await?,
                         )
                     } else {
+                        let r = match options.tree_shaking_mode {
+                            Some(TreeShakingMode::ReexportsOnly) => {
```

Comment:
Why only in this mode? Wouldn't this also be benefitial even when splitting into fragments?

Reply:
I think it will go to a different code path, namely `follow_reexports_with_side_effects`. 

https://github.com/vercel/next.js/blob/d0d73525a572216425c2e273e3edfcff2a60fa2b/turbopack/crates/turbopack-ecmascript/src/tree_shake/asset.rs#L219-L268

In the full module fragment tree shaking mode `import *` will not create `None` for `Option<ModulePart>`. The line producing `None` is located at https://github.com/vercel/next.js/blob/d0d73525a572216425c2e273e3edfcff2a60fa2b/turbopack/crates/turbopack-ecmascript/src/references/mod.rs#L633 which is only used for `ReexportsOnly`

Reply:
I see, I don't think we have a test for the other tree shaking mode. Could you add that, seems like we already have something similar going on for `mui-utils` which runs the same code from `side-effects-optimization/mui-utils` but just with a different config